lib/posix/options/pthread.c

```diff
@@ -384,6 +384,46 @@ int pthread_attr_setstack(pthread_attr_t *_attr, void *stackaddr, size_t stacksi
 	return 0;
 }
 
+/**
+ * @brief Get scope attributes in thread attributes object.
+ *
+ * See IEEE 1003.1
+ */
+int pthread_attr_getscope(const pthread_attr_t *_attr, int *contentionscope)
+{
+	struct posix_thread_attr *attr = (struct posix_thread_attr *)_attr;
+
+	if (!__attr_is_initialized(attr) || contentionscope == NULL) {
+		return EINVAL;
+	}
+	*contentionscope = attr->contentionscope;
+	return 0;
+}
+
+/**
+ * @brief Set scope attributes in thread attributes object.
+ *
+ * See IEEE 1003.1
+ */
+int pthread_attr_setscope(pthread_attr_t *_attr, int contentionscope)
+{
+	struct posix_thread_attr *attr = (struct posix_thread_attr *)_attr;
+
+	if (!__attr_is_initialized(attr)) {
+		LOG_ERR("Invalid pthread_attr_t or contentionscope");
+		return EINVAL;
+	} else if (contentionscope == PTHREAD_SCOPE_PROCESS) {
```

Comment:
`else` is unnecessary here due to the previous  return. Would it make sense to just move it to a separate `if` below?

Also, can you please use LOG_DBG instead of LOG_ERR?

Reply:
```
	if (!__attr_is_initialized(attr)) {
		LOG_DBG("attr %p is not initialized", attr);
		return EINVAL;
	}
	if (contentionscope == PTHREAD_SCOPE_PROCESS) {
		LOG_DBG("Value of contentionscope not supported %d", contentionscope);
		return ENOTSUP;
	}
	if (contentionscope != PTHREAD_SCOPE_SYSTEM) {
		LOG_DBG("Invalid contentionscope %d", contentionscope);
		return EINVAL;
	}
```